Hyperlink DOIs against preferred resolver

jpilgrim opened the pull request:
#1418, replacement of #932 by @katrinleinweber

Applied changes proposed by @katrinleinweber and did some additional bib cleanup.

@katrinleinweber Thank you very much for your support. Since we have these weird IP-validation problems and I did not want to bother you more than necessary, I re-applied you changes here. Please have a look -- if this is ok with you, let us know. I cannot assign you as a reviewer since you are not committer. Please simply let us know via a comment in this PR.

katrinleinweber commented:
Hello & thanks for resuscitating this :-) Reviewing the changes, I noticed that many `Bdsk-File` fields/blobs are changed here, but not in my PR.

How about this cleaner version: https://github.com/eclipse/n4js/compare/master...katrinleinweber:GH-1418-2?expand=1 I ran the script I used to replace the DOI links again, thus possibly capturing some newer links which cherry-picking or rebasing the commits from #932 did not contain.

jpilgrim commented:
@katrinleinweber Maybe there is one other link in your branch. When I replaced the links manually (well, that also was only a search-replace operation) I noticed some Bibdesk specific stuff in the bib file. I removed that to make the file bette maintainable for non-Bibdesk users (besides, the URLs I removed are internal links to the file system, not usable to anyone anymore). I will merge this PR now to close this issue. Thank you for your help!

katrinleinweber commented:
OK, cool :-) Thank you!
